packages/app-page-builder-elements/src/components/Elements.tsx

```diff
@@ -31,7 +31,10 @@ export const Elements: React.FC<ElementsProps> = props => {
     const elements = props.element.elements;
 
     let parentBlockElement: ElementType;
-    if (props.element.data.blockId) {
+    if (
+        props.element.data.blockId ||
+        (props.element.type === "block" && currentRendererMeta.parentElement.data.templateId)
+    ) {
```

Comment:
Please make this `if` statement a bit more readable:
1) extract individual conditions into local variables above the statement, like `isBlockReference` (or whatever that conditional does)
2) leave some comments on what each conditional is for, because the second one with `templateId` is unclear.

Keep in mind that this code will be maintained by someone, so inline comments would greatly help understand the intention.